server/src/main/java/org/elasticsearch/repositories/RepositoryData.java

```diff
@@ -134,13 +148,32 @@ public SnapshotState getSnapshotState(final SnapshotId snapshotId) {
         return indices;
     }
 
+    /**
+     * Returns the list of {@link IndexId} that have their snapshots updated but not removed (because they are still referenced by other
+     * snapshots) after removing the given snapshot from the repository.
+     *
+     * @param snapshotId SnapshotId to remove
+     * @return List of indices that are changed but not removed
+     */
+    public List<IndexId> indicesToUpdateAfterRemovingSnapshot(SnapshotId snapshotId) {
+        return indexSnapshots.entrySet().stream()
```

Comment:
In ES this got introduced in https://github.com/elastic/elasticsearch/commit/be397b7e0d892fd7f8e3e75308161f7fe0c4a56a do we have the other changes from that?

Reply:
All changes were already in, i took the related tests as well now.